underfs/ssh/src/main/java/alluxio/underfs/ssh/RemoteIdInfo.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.underfs.ssh;
+
+import alluxio.Configuration;
+
+import com.google.common.io.Closer;
+import net.schmizz.sshj.sftp.RemoteFile;
+import net.schmizz.sshj.sftp.SFTPClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * A class used to map remote user or group id to name.
+ */
+@ThreadSafe
+public class RemoteIdInfo {
+  private static final Logger LOG = LoggerFactory.getLogger(RemoteIdInfo.class);
+  private static final int MIN_REFRESH_INTERVAL = 5 * 60 * 1000;
+
+  private SFTPClient mSftpClient;
+  private String mRemotePath;
+  private long mLastRefreshTime = 0;
+
+  // This class is designed to be thread safe, and these maps are instantiated to
+  // be normal map. So the "modify" methods of maps should not be called.
+  private Map<Integer, String> mId2Name = new HashMap<>();
+  private Map<String, Integer> mName2Uid = new HashMap<>();
+
+  /**
+   * Construct a new {@link RemoteIdInfo}.
+   *
+   * @param sftpClient the client used to fetch remote file
+   * @param remotePath for which path to fetch
+   */
+  public RemoteIdInfo(SFTPClient sftpClient, String remotePath) {
+    mSftpClient = sftpClient;
+    mRemotePath = remotePath;
+  }
+
+  /**
+   * Guess the user name from id.
+   *
+   * @param id the id of the user
+   * @return the user name
+   */
+  public String guessName(Integer id) {
+    Map<Integer, String> map = mId2Name;
```

Comment:
why not refer to mId2Name directly?

Reply:
I will explain in comment of source code